zk/watcher.go

```diff
@@ -118,6 +124,9 @@ func (w *Watcher) reconnect() error {
 	}
 
 	go func() {
+		// QUESTION: why do we need to range over events? childrenW() also returns an event chanel.
```

Comment:
This is probably just because we ported the old c based library to the go native one about 2 years go.

Reply:
@scottjab-stripe We can clean up the code there. I checked  go-zookeeper. We will never get an event with event.Err != nil from the event channel returned by zk.Connect()